website/docs/reference/widgets/table/README.md

```diff
@@ -2,305 +2,316 @@
+    "status": "pending"
+  },
+  {
+    "step": "#3",
+    "task": "Bind the query using => fetch_users.data",
+    "status": "pending"
+  }
+]
+```
+
 
-<VideoEmbed host="youtube" videoId="-rzePEV2QQ8" title="How to use Table Widget" caption="How to use Table Widget"/>
 
-## Display data in tables
+### Display data dynamically
 
-To get data from a query to appear in a table widget, follow these steps:
+You can dynamically generate options by fetching data from queries or JS functions by binding the response to the **Table Data** property.
 
-1. If you don't already have a query that returns data from a datasource, [create one first](/core-concepts/data-access-and-binding/querying-a-database).
-2. Select your table widget to open its properties pane.
-3. In its properties pane, enter the following code snippet into its [**Table data** property:
-    ```javascript
-    // replace <query_name> with the name of your query
-    {{<query_name>.data}}
-    ```
+---
+**Example 1:** suppose you have data stored in a database and you want to display it in the Table widget.
 
-If you are doing work on your data in a [JS object](/core-concepts/writing-code/javascript-editor-beta) before it goes into the table and you need to supply the data from there, you should instead reference whichever of the JS object's properties returns your final results. For example, if you have a JS object called `utils` with a function `formatData`, you might put the following snippet into the table's **Table data** field:
+1.  Fetch data from the [sample database](https://docs.appsmith.com/core-concepts/connecting-to-data-sources/connecting-to-databases#sample-databases) `users` using a SELECT query `fetchData` to retrieve the data.
 
-```javascript
-{{ utils.formatData() }}
+```sql
+SELECT * FROM users ORDER BY id LIMIT 10;
 ```
 
-You can look at how this is set up in [this sample app](https://app.appsmith.com/applications/61e010e7eb0501052b9fa0f0/pages/61fba49b2cd3d95ca414b364).
-
-Now, the table widget should be populated with the data coming from the query. By default, this query also now runs automatically when the page loads. You can change this behavior from the query's settings page.
+2. In the Table's **Table Data** property, display the data using:
 
-### Transform table data
+```js
+{{fetchData.data}}
+```
+The Table widget displays all the data from the query. 
 
-Some API / Query responses might have deeply nested, unnecessary, or unhelpfully formatted fields. These can be transformed to fit your needs by parsing and processing the data with JavaScript. The JS [`map()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/map) function is highly recommended for processing raw data from your queries to fit your table correctly.
+---
+**Example 2:** if the data retrieved from the query is not in the desired format, you can use JavaScript to transform it before passing it to the Table widget.
```

Comment:
The current api example is more useful. Can we keep that as discussed during the review call?